packages/vite/src/node/utils.ts

```diff
@@ -622,7 +612,16 @@ function optimizeSafeRealPathSync() {
     safeRealpathSync = fs.realpathSync
     return
   }
-
+  // Check the availability `fs.realpathSync.native`
+  // in Windows virtual and RAM disks that bypass the Volume Mount Manager
+  // get the error EISDIR: illegal operation on a directory
+  try {
+    fs.realpathSync.native(path.resolve('./'))
+  } catch (error) {
+    if (~error.message.indexOf('EISDIR: illegal operation on a directory')) {
+      supportedRealpathSync = fs.realpathSync
```

Comment:
If we are going to use `fs.realpathSync` then we don't need the mapped version. We can directly assign to `safeRealPathSync` here and return. And remove the `supportedRealpathSync` variable.
```suggestion
      safeRealpathSync = fs.realpathSync
      return
```

Reply:
Thank you, I've made the corrections.